packages/gen-ai-hub/src/utils/deployment-cache.ts

```diff
@@ -51,20 +52,57 @@ function createDeploymentCache(cache: Cache<Deployment>) {
      * @param deployments - Deployments to retrieve the IDs and models from.
      */
     setAll: (
-      opts: Omit<DeploymentResolutionOptions, 'model'>,
+      opts: DeploymentResolutionOptions,
       deployments: AiDeployment[]
     ): void => {
       // go backwards to cache the first deployment ID for each model
-      [...deployments].reverse().forEach(deployment => {
-        cache.set(getCacheKey({ ...opts, model: extractModel(deployment) }), {
-          entry: transformDeploymentForCache(deployment)
+      const cacheEntries = [...deployments].reverse().map(deployment => ({
```

Comment:
@MatKuhr what do you think about that?

My 2 cents: semantically much better, implementation-wise more error prone and complicated to read. I am not convinced ;)